### ISSUE TITLE
Lambda function fails executing the binary as the file permissions are not preserved in the zip

### ISSUE THREAD

**Issue by krzysu** [serverless/serverless#504]
I have exactly the same issue as described here:
https://github.com/motdotla/node-lambda/issues/19

is short:

If you include a binary file in the package which is meant to be called by the Lambda function it will fail executing the binary as the file permissions are not preserved in the zip.

Lambda will raise the following error message:
```
{
  "errorMessage": "Process exited before completing request"
}
```
This is caused by the following error when calling execFile using the child_process package.
```
[Error: spawn EACCES] code: 'EACCES', errno: 'EACCES', syscall: 'spawn'
```

I'm using ubuntu, if I zip component folder and upload to s3 manually and then load my lambda function from that s3 .zip file, all executes correctly.

**Comment by mikedizon** [comment]
+1

**Comment by mikedizon** [comment]
Wonder if archiverjs would solve this.

**Comment by perhallstroem** [comment]
+1 

We seem to be having the same problem. Porting existing functions to serverless and getting an errno 13 when trying to execute a binary.

**Comment by fredgreer** [comment]
+1 here too

**Comment by mikedizon** [comment]
Wish this could be prioritized somehow. I'd address it myself but don't know enough about the core

**Comment by shox** [comment]
+1 same issue

**Comment by mewalig** [comment]
+1

**Comment by krzysu** [comment]
6 votes so far, @ac360 can we get some attention here? :smiley:

**Comment by flomotlik** [comment]
resolved in #1631 for Serverless 1.0. Closing this issue therefore
